protocol/messenger.go

```diff
@@ -844,6 +844,8 @@ func (m *Messenger) Start() (*MessengerResponse, error) {
 		return nil, err
 	}
 
+	m.checkForMissingMessages()
```

Comment:
Do you think we could make it like this:
```suggestion
	go m.checkForMissingMessagesLoop()
```

... and remove the `go func() { ... } ()` wrapper inside the function itself? 
It will make it more readable.